### ISSUE TITLE
OnComplete event called more than once for React JS

### ISSUE THREAD

**Issue by svemula09** [surveyjs/surveyjs#274]

**Comment by andrewtelnov** [comment]
@svemula09 What version do you use? I just checked it out and it calls for me one time.
Probably some options may lead to the issue. If you are using the latest version, could you share your survey JSON and JavaScript code with us?

Thank you,
Andrew

**Comment by svemula09** [comment]
My Code: 

 `import React, {PropTypes} from 'react';
import * as Survey from 'survey-react';
import {browserHistory} from 'react-router';
import {Modal} from 'react-bootstrap';
import {Segment} from 'semantic-ui-react';
import {connect} from "react-redux";
import {bindActionCreators} from "redux";
import * as PersonalSuggestionsActions from './PersonalSuggestions.actions';

class PersonalSuggestionsPage extends React.Component {

   constructor() {
    super();

    this.surveyJSON =
    {
      pages: [
        {
          name: "InitialInformation1",
          questions: [
            {
              type: "text",
              name: "age",
              title: "Age",
              isRequired: true,
              validators: [
                {
                  type: "numeric",
                  text: "Please choose an age ranging from 10 - 120",
                  minValue: "10",
                  maxValue: "120"
                }
              ],
              inputType: "number",
              placeHolder: "age"
            },
            {
              type: "radiogroup",
              name: "relationship",
              title: "Relationship status",
              isRequired: true,
              choices: [
                {
                  value: "single",
                  text: "Single"
                },
                {
                  value: "married",
                  text: "In a relationship"
                }
              ]
            }
          ]
        },
        {
          name: "InitialInformation2",
          questions: [
            {
              type: "radiogroup",
              name: "children",
              title: "Do you have children?",
              isRequired: true,
              choices: [
                {
                  value: "yes",
                  text: "Yes"
                },
                {
                  value: "no",
                  text: "No"
                }
              ]
            }
          ]
        },
        {
          name: "InitialInformation3",
          questions: [
            {
              type: "radiogroup",
              name: "adultDependents",
              title: "Do you have adult dependents?",
              isRequired: true,
              choices: [
                {
                  value: "yes",
                  text: "Yes"
                },
                {
                  value: "no",
                  text: "No"
                }
              ]
            }
          ]
        }
      ]
    };
    this.survey = new Survey.Model(this.surveyJSON);
    this.survey.showProgressBar = "top";
    this.customCSS = {
      "root": "",
      "header": "panel-heading",
      "body": "panel-body",
      "footer": "ui basic left-padded",
      "navigationButton": "ui  button mBackground",
      "navigation": {
        "complete": "",
        "prev": "",
        "next": ""
      },
      "progress": "ui progress warning",
      "progressBar": "m-bar",
      "pageTitle": "",
      "row": "",
      "question": {
        "root": "",
        "title": "",
        "comment": "form-control",
        "indent": 20
      },
      "error": {
        "root": "alert alert-danger",
        "icon": "glyphicon glyphicon-exclamation-sign",
        "item": ""
      },
      "checkbox": {
        "root": "form-inline",
        "item": "checkbox",
        "other": ""
      },
      "comment": "form-control",
      "dropdown": "form-control",
      "matrix": {
        "root": "table table-striped"
      },
      "matrixdropdown": {
        "root": "table"
      },
      "matrixdynamic": {
        "root": "table",
        "button": "button"
      },
      "multipletext": {
        "root": "table",
        "itemTitle": "",
        "itemValue": "form-control"
      },
      "radiogroup": {
        "root": "form-inline",
        "item": "radio",
        "other": ""
      },
      "rating": {
        "root": "btn-group",
        "item": "btn btn-default"
      },
      "text": "form-control",
      "window": {
        "root": "modal-content",
        "body": "modal-body",
        "header": {
          "root": "modal-header panel-title",
          "title": "pull-left",
          "button": "glyphicon pull-right",
          "buttonExpanded": "glyphicon pull-right glyphicon-chevron-up",
          "buttonCollapsed": "glyphicon pull-right glyphicon-chevron-down"
        }
      }
    };

    this.sendDataToServer = this.sendDataToServer.bind(this);
  }


  sendDataToServer(survey) {
    debugger;
    this.props.handleOnHide(false);
    const surveyData = Object.assign({}, survey.data);
    const resultAsString = JSON.stringify(surveyData);
    window.dataLayer = {
      "questionaireResults" : resultAsString
    };

    console.log(resultAsString);
    this.props.actions.onQuestionnaireCompletion(surveyData)
      .then(() => { browserHistory.push('/recommendations/products'); });
  }

  render() {
    return (
      <Modal show={this.props.show} onHide={this.props.onHide} aria-labelledby="contained-modal-title-sm">
        <Modal.Header closeButton>
          <h2>Tell us about yourself</h2>
        </Modal.Header>
        <Modal.Body >
          <Segment basic id="QUESTIONNAIRE_FORM">
            <Survey.Survey model={this.survey} onComplete={this.sendDataToServer}  css={this.customCSS}/>
          </Segment>
        </Modal.Body>
      </Modal>
    );
  }
}

PersonalSuggestionsPage.propTypes = {
  state: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired,
  handleOnHide: PropTypes.func.isRequired
};

PersonalSuggestionsPage.defaultProps = {};

function mapStateToProps(state, ownProps) {
  return {
    state: state
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(PersonalSuggestionsActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(PersonalSuggestionsPage);
`

**Comment by svemula09** [comment]
File attached. Please verify - Thank you for responsive so quickly


[PersonalSuggestionsPageCopy.txt](https://github.com/surveyjs/surveyjs/files/815466/PersonalSuggestionsPageCopy.txt)

**Comment by tsv2013** [comment]
@svemula09 Thank you for provided code.
I've tried your surveyJSON, but in our example `onComplete` is called only once.

I've created react+redux+survey sample application to test your code. But have no success:
- some inports are absent (import * as PersonalSuggestionsActions from './PersonalSuggestions.actions';)
- after component created I got an empty page w/o any errors

I attached sample project: [sample.zip](https://github.com/surveyjs/surveyjs/files/816131/sample.zip)
